### ISSUE TITLE
VisibilityModifierCheck: docuement not handling top-level fields in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20270]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/design/visibilitymodifier.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="VisibilityModifier"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputVisibilityModifierCompact.java <<'EOF'
public int publicField = 1;
protected int protectedField = 2;
int packageField = 3;

void main() { }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.6.0-SNAPSHOT-all.jar -c config.xml InputVisibilityModifierCompact.java

Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**

VisibilityModifierCheck must flag every non-private top-level field in a compact source file. Top-level fields are members of the implicit class and the check's contract is "non-private instance fields require accessors". The implicit class being in the unnamed package does not weaken this rule, accidental public state is exactly what the check exists to prevent.

Expected on the compact input: three violations, one each on `publicField`, `protectedField`, and `packageField`.

Root cause: `isAnonymousClassVariable` at `VisibilityModifierCheck.java:337-338` returns `variableDef.getParent().getType() != TokenTypes.OBJBLOCK`. The helper uses "parent is not `OBJBLOCK`" as a proxy for "field of an anonymous inner class" to skip those. For a top-level field in a compact source file the parent is `COMPACT_COMPILATION_UNIT`, so the helper returns `true` and every top-level field is silently excluded.

**Same code inside an ordinary class:**

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > InputVisibilityModifierOrdinary.java <<'EOF'
public class InputVisibilityModifierOrdinary {
    public int publicField = 1;
    protected int protectedField = 2;
    int packageField = 3;

    public void main() { }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.6.0-SNAPSHOT-all.jar -c config.xml InputVisibilityModifierOrdinary.java

Starting audit...
[ERROR] InputVisibilityModifierOrdinary.java:2:16: Variable 'publicField' must be private and have accessor methods. [VisibilityModifier]
[ERROR] InputVisibilityModifierOrdinary.java:3:19: Variable 'protectedField' must be private and have accessor methods. [VisibilityModifier]
[ERROR] InputVisibilityModifierOrdinary.java:4:9: Variable 'packageField' must be private and have accessor methods. [VisibilityModifier]
Audit done.
Checkstyle ends with 3 errors.
```

**Comment by arturict** [comment]
I am on it.

**Comment by romani** [comment]
fix is merged.
